### ISSUE TITLE
Cannot make the .pb predict at all

### ISSUE THREAD

**Issue by eladmeir** [PINTO0309/onnx2tf#507]
### Issue Type

Documentation Feature Request

### OS

Linux

### onnx2tf version number

1.16.29

### onnx version number

1.14.0

### onnxruntime version number

1.15.1

### onnxsim (onnx_simplifier) version number

0.4.33

### tensorflow version number

2.13.0

### Download URL for ONNX

https://github.com/Megvii-BaseDetection/YOLOX/releases/download/0.1.1rc0/yolox_s.onnx

### Parameter Replacement JSON

```json
I could figure out if one is needed
```

### Description

1. Research
2. I wish to use the .pb file but cannot understand how
3. onnx2tf -i yolox_s.onnx -otfv1pb
4. because I love plain TensorFlow (TFLite works great, but I wish to use TF)
5. https://github.com/Megvii-BaseDetection/YOLOX - the source for the onnx model

**Comment by PINTO0309** [comment]
Can you tell me exactly what the problem is?

yolox_s_float32.pb

![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/8e44cf24-e2c8-42b7-a66d-67f161e59a77)

![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/84699cc2-e1fb-458f-b3e2-7fa28fab3bbb)

**Comment by eladmeir** [comment]
Hey, Thanks so much for the quick reply

I saw the implementation, and the TFLite is indeed working great... That's why I am asking - it seems like a very trivial solution that I am not aware of..

The exact problem is that I am trying to predict with the following code, and I think that I am missing something trivial with the work of Graphs, maybe you could assist me...

`def one_more_copa(model_path='aph.pb'):
    import tensorflow as tf
    from tensorflow.python.platform import gfile

    x_org = cv2.imread('72b.jpg')
    x_org = cv2.resize(x_org, (480, 480)).copy()
    x = x_org.astype('float32')
    with tf.compat.v1.Session(graph=tf.Graph()) as sess:
        graph_def = tf.Graph().as_graph_def()
        graph_def.ParseFromString(open(model_path, 'rb').read())
        # graph_def = tf.GraphDef()
        # graph_def.ParseFromString(f.read())
        sess.graph.as_default()
        g_in = tf.import_graph_def(graph_def)
        tensor_output = sess.graph.get_tensor_by_name('images:0')
        tensor_input = sess.graph.get_tensor_by_name('Identity:0')
        predictions = sess.run(tensor_output, {tensor_input: x})
        print(predictions)
`

And I get this error:
`python3.8/site-packages/tensorflow/python/framework/ops.py", line 3766, in _as_graph_element_locked
    raise KeyError("The name %s refers to a Tensor which does not "
KeyError: "The name 'images:0' refers to a Tensor which does not exist. The operation, 'images', does not exist in the graph."

Process finished with exit code 1`

**Comment by PINTO0309** [comment]
Ah, I hadn't tested in .pb in about 4 years, so I forgot about it. I'll add `:0` to the input/output and test it.

**Comment by eladmeir** [comment]
Great to hear that you are on board, I would be very glad to assist if I could, maybe share my trials?
Let me know if I could help

**Comment by PINTO0309** [comment]
I had a hard time remembering the logic for v1, but I did get the output.
```python
import tensorflow as tf
tf.compat.v1.disable_v2_behavior()
import numpy as np

x_org = np.ones((1,640, 640,3))
x = x_org.astype('float32')

def get_graph_def_from_file(graph_filepath):
    with tf.compat.v1.gfile.GFile(graph_filepath, 'rb') as f:
        graph_def = tf.compat.v1.GraphDef()
        graph_def.ParseFromString(f.read())
        return graph_def

with tf.compat.v1.Session(graph=tf.compat.v1.Graph()) as sess:
    graph_def = get_graph_def_from_file('yolox_s_float32.pb')
    tf.import_graph_def(graph_def, name='')
    tensor_input = sess.graph.get_tensor_by_name('images:0')
    tensor_output = sess.graph.get_tensor_by_name('Identity:0')
    predictions = sess.run(tensor_output, {tensor_input: x})
    print(predictions)
```
```
[[[ 2.3274002e+00  6.8202913e-01  1.3630362e+00 ...  1.9085901e-03
    1.3318274e-03  1.6112461e-03]
  [ 2.7896819e+00  6.9194186e-01  2.2075419e+00 ...  7.4822188e-04
    5.4520078e-04  5.0522562e-04]
  [ 2.5516617e+00  7.5139928e-01  2.3757622e+00 ...  4.4952985e-04
    2.8451657e-04  2.9596273e-04]
  ...
  [ 2.5698388e-01 -6.8696976e-01  1.1723595e+00 ...  1.8904940e-03
    1.2838489e-03  1.6660128e-03]
  [ 1.2140840e-01 -5.2784765e-01  1.1206152e+00 ...  8.3787512e-04
    5.2379118e-04  8.3001301e-04]
  [-7.1434128e-01 -1.1008520e+00  1.2664701e+00 ...  1.8842940e-03
    1.6071482e-03  1.8508373e-03]]]
```

**Comment by PINTO0309** [comment]
No reply, so I will close it.

**Comment by eladmeir** [comment]
Sorry for the late reply, we've Had Yom Kippur and didn't get to my computer up until today

Thank you very much for your assistance, that was perfect!

**Comment by eladmeir** [comment]
Hey @PINTO0309 I have found a nice way of making this converted model run with a `@tf.function`, and without the session scope, which make it suitable for OOP, and enable TF v2 capabilities for inference
This one works nice with nicely with TF 2.4.1 **(INFERENCE)**, while to model was created with onnx2tf==1.17.5, and TF 2.13.1 **(TRAINING)** and it goes like this

`
class Predictor(object):
    def __init__(
        self,
        model_checkpoint_path=None
    ):
        self.model = self.init_model(model_checkpoint_path, test_size)

    @staticmethod
    def init_model(model_checkpoint, test_size):
        from tensorflow.core.framework.graph_pb2 import GraphDef
        import tensorflow as tf

        @tf.function
        def my_model(x):
            gd = GraphDef()
            with open(model_checkpoint, 'rb') as f:
                gd.ParseFromString(f.read())
                y = tf.graph_util.import_graph_def(gd, name='', input_map={<YOUR_INPUT_LAYER_NAME>: x},
                                                   return_elements=['<YOUR_OUTPUT_LAYER_NAME>:0'])
            return y

        x = tf.keras.Input(shape=(test_size[0], test_size[1], 3))
        print(test_size)
        y = tf.keras.layers.Lambda(my_model)(x)
        model = tf.keras.Model(x, y)
        return model

def predict(self, img):
    return self.model(tf.constant(img))
`

I would be more than happy to contribute to this wonderful repo, if you want me to add my take on inference with TF
Please let me know if you are willing to add it

Thank you very much for your amazing assistance

**Comment by PINTO0309** [comment]
I see. That's interesting.

>I would be more than happy to contribute to this wonderful repo, if you want me to add my take on inference with TF
Please let me know if you are willing to add it

Yes, I might be happy to get a pull request for a tutorial as a sample inference somewhere in the README. I have quite a bit of documentation in the README, so you may have a very hard time issuing a pull request.

**Comment by eladmeir** [comment]
Yes, I have noticed the fact that your README is quite comprehensive

Maybe we could start off a "demo" section for usage examples? From my experience those are great for ramping up the usage numbers, and make it easier for beginners to take part, if that's something you are willing to advance in this repo.
Let me know if you are up for it and I could make the first step with this end to end demo of taking a YoloX model, which was trained with pytorch and converted to onnx, and demonstrate how to make a saved model conversion, and TF1+TF2 inference options 

Waiting for your approval.

**Comment by PINTO0309** [comment]
Thank you.

Fortunately, I believe that Pull Request is a mechanism to improve the ideas you suggested while visualizing them and exchanging opinions with each other, so it would be easier for us to discuss once you show us the image of the improvement of README.

My important policy in creating this tool is, "Don't start out perfect, but improve little by little." I rarely care if I notice that we've made a little bit of a mistake, because if I notice that we've made a mistake, I fix it right then and there and release it right away.

Don't hesitate. This tool is not a large tool made by a large company. It is a personal tool that I am making almost by myself. :+1:
